crates/sc-consensus-subspace/src/archiver.rs

```diff
@@ -356,12 +358,97 @@ pub struct ObjectMappingNotification {
     // TODO: add an acknowledgement_sender for backpressure if needed
 }
 
+/// When to start creating object mappings.
+///
+/// There is no setting for starting from the last archived block, because we can't guarantee that
+/// all the mappings from that block were received by clients before the node shut down. Instead,
+/// clients should tell the node to create mappings from the latest piece index they know about.
+#[derive(Copy, Clone, Debug, Default, Eq, PartialEq)]
+pub enum CreateObjectMappings {
+    /// Start creating object mappings from this segment index.
+    Segment(SegmentIndex),
```

Comment:
Hmm, that’s an interesting question. If you just have a list of mappings, how will you know which blocks they’re from?

I was expecting the client/user to take the last piece index from their mappings, and supply it on the command line. I could change this type to `PieceIndex` to make that clearer?
(And then do the checks against the segment index for that piece, because that’s what we have in the code.)

I’ll ask Carlos what he’s expecting to do for the implementation.

Reply:
I just had an idea about how to do this:

We add the block number to each subscription response, at the top level. So each response would look like:
- block number
- list of mappings

The client could then track the latest block number it is up to, but wouldn’t need any earlier block numbers. (Because block numbers are not required to fetch objects, just for recovery after downtime.)

This wouldn’t add much to the size of the response, because there’s only one block number for each block with mappings.

We’d just have to repeat the block number if we split a block’s mappings between batches, but that seems ok.

Reply:
Makes sense